### PR TITLE
Question redaction

### LINKED ISSUE #75: There should be a way to redact a particular wiki survey
We should have a way to redact a particular wiki survey by overwriting the question text and all ideas with some message about being deleted or redacted.

Initially, this can be a script that we run manually when requested by the creators of a wiki survey.

### PR DESCRIPTION
**Pull request by lukebaker**
Fixes https://github.com/allourideas/allourideas.org/issues/75

This PR adds a method for redacting a question (`redact!`) and a rake task that will run that method if given a question_id. Redacting a question will replace the question text and all choice text with "Redacted".

### CODE REVIEW

**Comment by msalganik** [comment]
Thanks @lukebaker.  I can't really review the code, but I like the idea that is done by question_id not the user email address.

One question:
- should we redact the wiki survey creator's email address as well?  I think yes, but I'm open to other ideas.

**Comment by lukebaker** [comment]
> should we redact the wiki survey creator's email address as well? I think yes, but I'm open to other ideas.

I don't think we expose that publicly anywhere, so I didn't think we needed to do that. However, I could see the case for also disabling the wiki survey and removing it from the creator's list of wiki surveys. We can chat more about that in our meeting next week, if you'd like.

**Comment by msalganik** [comment]
OK, let's chat about it.  When people ask to have their information deleted, I think they might mean their email addresses too.

File: app/models/question.rb [hunk 1]
```diff
@@ -44,6 +44,16 @@ def pick(algorithm = nil)
   named_scope :created_by, lambda { |id|
     {:conditions => { :local_identifier => id } }
   }
+  REDACTED_TEXT = "Redacted"
```
Comment (lukebaker) [inline app/models/question.rb:47]:
```suggestion
  REDACTED_TEXT = "Redacted at request of wiki survey owner"
```

File: spec/models/question_spec.rb [hunk 2]
```diff
@@ -252,6 +252,26 @@
...
+    before(:all) do
+      truncate_all
+      @q = Factory.create(:aoi_question)
+    end
+
+    it "should redact all choices and question text" do
+      @q.redact!
+      @q.name.should == Question::REDACTED_TEXT
+      latest = @q.versions.latest
+      latest.previous.name.should_not == Question::REDACTED_TEXT
```
Comment (lukebaker) [inline spec/models/question_spec.rb:265]: Overwrite all prior versions with redacted text.

**Comment by lukebaker** [comment]
Remove my account:
* Redact email address
* Reset password
* Redact all wiki surveys

On redact of wiki survey:
* Disable wiki survey
* Transfer wiki survey to Matt's account

**Pull request merged**